### DISCUSSION TITLE
AI art depth maps for laser engraving and CNC tooling use cases.

### DISCUSSION THREAD

**Discussion by El7ias** [thygate/stable-diffusion-webui-depthmap-script#73] [Ideas]
I'm so thrilled Depth maps are now a part of the SD1111. This is big! THANK YOU! I have been pursuing depth maps the moment I found Ai art. I went so far as to start making my own training model, then I found out about MIdas. Fantastic! 

Okay here is my use case.  I have a long background in 3D modeling and CAD. I have been making depth maps from my own models to then send to my fiber laser for 2.5D reliefs as well as sending the STLs into CAD/CAM for cnc routing. loads of Maker fun! 

The maps I have been able to make using Depth map script is promising but lacks the detail in the compositions to really POP a design's look. I am hoping the models get better soon so the really fine details of an image really come through with depth. Please help this happen, there is so much to make and teach others. 

[The art](https://drive.google.com/file/d/1uSmB8AbKw5cexHIRaJ5MME6zc0TKQFnE/view?usp=sharing)
[The edited Midas map](https://drive.google.com/file/d/1iDjH_fcx3lvGQKfSrr7NOpb_Tsn4FGbQ/view?usp=sharing)
[The laser engraving - I had to modify the depth map in photoshop, not great.](https://drive.google.com/file/d/1UFiCR_f_9ajkK4LHmRTeRG_H5xVQNpAt/view?usp=sharing)

**Comment by thygate** [comment]
That is a very nice result, I wonder how much manual massaging it required in the various stages it went through ;) Very impressive !

There's also the https://github.com/compphoto/BoostYourOwnDepth repo that allows for manual merging of depthmaps. I might implement this in the future.

This extension now includes support for the midas 3.1 models, including the one trained on 512x512, beware, it's very vram hungry.
In my personal experience, I find the results from leres to be more detailed (and faster, and uses less memory) compared to the midas models, even though it might not be as accurate at times. (See observations in readme)

I'm no stranger to CNC or CAD/CAM, I spent quite a few years writing machine controller software. And I spent a lot of evenings at the bar with the guy writing the toolpath generation software, what software are you using to generate toolpaths from a depthmap ? Are you displacing a plane first, and then use that in a 3D CAM package ? I've been out of the game for a while now, and most (big!) software I was familiar with now seems defunct :\

I mostly only 3D print now, and outsource CNC and laser work ..

**Reply by El7ias** [reply]
For Laser etching I use LightBurn for Galvo. it uses heightmaps to generate 2.5d ablation of layers to get a relief, in my case removing metal and stone. The more detailed a heightmap the better the quality and I will sacrifice vram for quality every time.  For CNC carving in other materials, I can convert the heightmap to an STL, again more detail = better quality, and import that into Fusion 360 and generate tool passes that I can then export out to my old Mach3 machine. Same goes for 3D printing, so this is why the continued refinement of AI generated depth maps have massive potential in my and so many others' workflows. We love it! We need it. 

As for the Medusa that I linked, to get to this test piece I had to manually edit the depth maps I made originally using a standalone Gradio UI and photoshop and some tools I use for 3D texture maps called ShaderMaps. It was a very lengthy, very experimental process and I honestly think the final results of the aluminum relief is not great. The dress and foreground elements lost a lot of details and are very shallow. 1st tests were nearly altogether flat. 

The very next day I found the Depth Map script for A1111 and have been obsessed with it since.  Thanks for the link to BoostYourOwnDepth, that is very interesting. I will deeply explore this.  Was wondering if I could train my own Textural Inversions or models that would work in A1111.  I would love to start developing for this however I can. 

Thank you!

**Reply by El7ias** [reply]
Oh yah, I also use Zbrush to quickly visualize the results of the generated maps. I Import the map as an Alpha then apply it to the face of a [cube with a dense mesh](https://drive.google.com/file/d/14xRVD5x-QJd4A2KPI9_2ebDvkYvykl9Z/view?usp=sharing). This helps let me know how to edit the map. The link is an updated version, more rounded features.

**Reply by thygate** [reply]
>I will sacrifice vram for quality every time

Yes, the aim is to have the best possible quality, but it's more a matter of also having the needed vram, and not running out.

>I had to manually edit the depth maps

That's why I brought up the BoostYourOwnDepth repo.

The Mach3 software was parallel port "dir and step" steppermotor driver only last time i encountered it .. That still the case ?

I really do like the Medusa result, feel free to share more examples =)

Commercial software like ZBrush is out of the picture here, OSS software only. It looks great though ..

Yes, check the A1111 webui [wiki](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#textual-inversion) for instructions on how to use textual inversion or dreambooth etc ..

**Reply by El7ias** [reply]
I only use zbrush as a way to quickly visualize the quality of the map. not for 3D modeling. It tells me how it would look if I imported it into my laser software and how I should edit it. 

You are correct about Mach3. There are other tools that use USB but most everything is a stepper motor on the other side.  

Thank you very much for the BoostYourOwnDepth repo link. That will be very helpful.

I'll post pics of the art and progress of the projects. Hope it's helpful and inspires 
thank you very much
